models/User.js

```diff
@@ -0,0 +1,38 @@
+import { mongoose, Schema }  from 'mongoose'
+
+const userSchema = new Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+
+  email: {
+    type: String,
+    required: true,
+    unique: true
+  },
+
+  role: {
+    type: Number,
+    enum: [0,1], // 0 = Resident, 1 = Admin
+    required: true
+  },
+
+  address: {
+    type: String,
+    required: function() {
+      return this.role === 0;
+    }
+  },
+
+  phoneNumber: {

```

Comment:
Nice job making the phone number a string data type, rather than a number! This will not only save room in the database (that's cool too) but most importantly make the data easier to handle!